### PR TITLE
Config overhaul

### PR DESCRIPTION
**Pull request by gravitypriest**

### CODE REVIEW

**Comment by kylape** [comment]
Got some py2.6 failures:

```
insights/client/__init__.py:12: in <module>
    from .config import InsightsConfig
E     File "/tmp/workspace/sights-core_config-overhaul-EDPJY5THITRKFP3PFYFDLKCLX4XR65B74QJ5XGQQYMAV2TYMO64Q/insights/client/config.py", line 342
E       DEFAULT_KVS = {k: v['default'] for k, v in DEFAULT_OPTS.iteritems()}
E                                        ^
E   SyntaxError: invalid syntax
```
dict comprehensions are not a thing in py2.6 :(

File: insights/client/containers.py [hunk 1]
```diff
@@ -71,13 +70,13 @@ def run_command_very_quietly(cmdline):  # shhhhhhhh
...
     UseDocker = False
 
-# force atomic or docker
-if config['use_docker']:
-    UseAtomic = False
-    UseDocker = True
-if config['use_atomic']:
-    UseAtomic = True
-    UseDocker = False
+# # force atomic or docker
```
[the remark this answers is not in the document]
Comment (gravitypriest) [reply insights/client/containers.py]: Meant to take this out actually. It's trying to access the global `config` object which is no longer valid. I guess I mistakenly broke these options so I'll revise this as well.

**Comment by gravitypriest** [comment]
RIP dict comp

And yeah I'll go over it and change everything to `getattr` syntax.

**Comment by kylape** [comment]
looks like we got some errors with the scheduling tests. mind to try running pytest on your local box and fixing whatever comes up?

**Pull request merged**